### ISSUE TITLE
qwen3 30BA3B faild and memory leak

### ISSUE THREAD

**Issue by xunbu** [lmstudio-ai/lmstudio-bug-tracker#645]
**Which version of LM Studio?**
Example: LM Studio 0.3.15 and  LM Studio 0.3.9

**Which operating system?**
Windows
8840hs APU.

**What is the bug?**
1.When I use qwen3 30B A3B, it fails to work properly if the context is slightly larger—memory usage keeps increasing until it crashes, and Eject doesn't help. This only happens when using vulkan llama.cpp, but not with cpu llama.cpp. 
2. When I back to  LM Studio 0.3.9, the memory leak disappear but still fail to response.

**Screenshots**

![Image](https://github.com/user-attachments/assets/d24a2a24-d13a-4ae7-9f26-9e28511dce86)

**Logs**
2025-05-06 16:26:55 [DEBUG] 
Sampling params:	repeat_last_n = 64, repeat_penalty = 1.000, frequency_penalty = 0.000, presence_penalty = 0.000
	dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
	top_k = 20, top_p = 0.800, min_p = 0.010, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.700
	mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
2025-05-06 16:26:55 [DEBUG] 
sampling: 
logits -> logit-bias -> penalties -> dry -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist 
generate: n_ctx = 4096, n_batch = 512, n_predict = -1, n_keep = 829
2025-05-06 16:26:55 [DEBUG] 
BeginProcessingPrompt
2025-05-06 16:26:55 [DEBUG] 
llama.cpp abort:5263: GGML_ASSERT(nei0 * nei1 <= 3072) failed

**Comment by xunbu** [comment]
i just see the solution in  #635 , but is the memory leak a bug? if not, the issue can be closed.

**Comment by waltercool** [comment]
> i just see the solution in [#635](https://github.com/lmstudio-ai/lmstudio-bug-tracker/issues/635) , but is the memory leak a bug? if not, the issue can be closed.

It's not the same issue. I can also reproduce this problem, and my only way to free memory is by killing node thread using lot of GPU memory (amdgpu_top)

It affects Vulkan llama.cpp v1.31.1, v1.31.0, but works fine with 1.29.0.

Using AMD Strix Halo 8060S

**Comment by waltercool** [comment]
Issue still happens with Vulkan llama..cpp v1.32 and v1.33

**Comment by waltercool** [comment]
And this is still something at v1.38.0.

Last working version seems to be v1.29.0.

Also, 1.38.0 has terrible performance, then crash.
